### PR TITLE
Notifications ngxs refactor

### PR DESCRIPTION
**Pull request by kphutiyagae**
# Description

Implemented NGXS state management for Notifications feature.

Fixes # 151-refactor-code-for-pubsub

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration.

- [x] Unit Tests
- [x] Component Tests

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: libs/app/notifications/data-access/src/notifications.actions.ts
Comment (ryanbasiltrickett) [inline libs/app/notifications/data-access/src/notifications.actions.ts:1]: This file should be in the utils folder under notifications

File: libs/app/notifications/utils/src/interfaces/notifications.interface.ts
Comment (ryanbasiltrickett) [inline libs/app/notifications/utils/src/interfaces/notifications.interface.ts:1]: Please send a small documented overview of these interfaces such that they may be integrated and used else where and approved

**Review by ryanbasiltrickett — CHANGES_REQUESTED** [review]

File: libs/app/notifications/data-access/src/notifications.actions.ts
Comment (kphutiyagae) [reply libs/app/notifications/data-access/src/notifications.actions.ts:1]: Done

File: libs/app/notifications/utils/src/interfaces/notifications.interface.ts
Comment (kphutiyagae) [reply libs/app/notifications/utils/src/interfaces/notifications.interface.ts:1]: Will be sent.

**Review by ryanbasiltrickett — APPROVED** [review]

**Pull request merged**